server/stycoon/command_trade.go

```diff
@@ -26,6 +26,10 @@ func (game *Game) processTrade(id int64, playerOwnedShips map[int64]struct{}, pl
 	_, shipOwnedByPlayer := playerOwnedShips[id]
 	_, targetIsPlanet := planetIDs[id]
 
+	if !shipOwnedByPlayer && !targetIsPlanet {
+		return fmt.Errorf("trading with another players ship is forbidden")
+	}
+
```

Comment:
I've moved this up because it seems like more important error than checks that we do later in `CheckPlanetCanTrade` and `CheckShipCanTrade`